File: golang/distributecandies575/distribute-candies-575.go
package main

import "fmt"

func main() {
	candies := [...]int{1, 1, 2, 3}
	fmt.Println(distributeCandies(candies[:]))
}

func distributeCandies(candies []int) int {
	set := make(map[int]bool)
	for _, v := range candies {
		set[v] = true
	}
	return min(len(set), len(candies)/2)
}

func min(x, y int) int {
	if x > y {
		return y
	} else {
		return x
	}
}

File: nodejs/distribute-candies-575.js
/**
 * @param {number[]} candies
 * @return {number}
 */

 'use strict';
var distributeCandies = function(candies) {
    let set = new Set();
    for (let v of candies) set.add(v);
    return Math.min(candies.length/2, set.size);
};

let x = distributeCandies([1,1,2,3])
console.log(x)

